lib/src/store.dart

```diff
@@ -58,22 +63,36 @@ class Store {
   /// As an example, [transformer] could be used to throttle the number of
   /// triggers this [Store] emits for state that may update extremely frequently
   /// (like scroll position).
-  Store.withTransformer(StreamTransformer<dynamic, dynamic> transformer) {
-    _streamController = new StreamController<Store>();
+  Store.withTransformer(StreamTransformer<dynamic, dynamic> transformer)
+      : _streamController = new StreamController<Store>() {
+    manageStreamController(_streamController);
 
     // apply a transform to the stream if supplied
     _stream = _streamController.stream
         .transform(transformer as StreamTransformer<Store, dynamic>)
```

Comment:
@evanweible-wf why do we accept `<dynamic, dynamic>` in the ctor and then cast it to `<Store, dynamic>`, why not just accept that from the start? Also, doesn't it actually need to be `<Store, Store>`?